libdevcore/Worker.cpp

```diff
@@ -48,9 +48,13 @@ void Worker::startWorking()
 			while (m_state != WorkerState::Killing)
 			{
 				WorkerState ex = WorkerState::Starting;
-				bool ok = m_state.compare_exchange_strong(ex, WorkerState::Started);
+				{
+					// When somebody waits for this change on m_state, they'll give me the lock saying 'm_state_notifier.wait(l)'
+					Guard l(x_work);
+					bool ok = m_state.compare_exchange_strong(ex, WorkerState::Started);
```

Comment:
I think we can just ignore the returned value instead of `(void)ok;`, no?

Reply:
Should be the same thing.  Probably it's just fine to `m_state = Started;`.